src/twisted/internet/endpoints.py

```diff
@@ -1466,6 +1473,13 @@ def _parseSSL(
         dhParameters=dhParameters,
         **kw,
     )
+    if cipher:
+        try:
+            cipherBytes = cipher.replace(",", ":").encode("ascii")
+            cf.getContext().set_cipher_list(cipherBytes)
```

Comment:
I guess that this will work.

By understanding is that context factories might not cache the context.

This is why, we might need to change the API to something like this ... but this will break all existing context factory implementation.

So I guess that this is somehow ok for now.

We can refactor once someone nees to used this code with a custom context factory.

 
```
            cf.setOpenSSLCipherList(cipherStr)
```

Reply:
@adiroiban what do you suggest?
Should we go ahead with making changes to the API ?